strawberry_django/type.py

```diff
@@ -1,45 +1,147 @@
+import dataclasses
+import django
 import strawberry
-import typing
-from .types import get_model_fields, update_fields
-from .utils import deprecated
+from django.db import models
+from strawberry.arguments import UNSET, is_unset
+from typing import Any, Optional
+
+from .fields.field import StrawberryDjangoField
+from .fields.types import (
+    is_optional,
+    get_model_field, resolve_model_field_type, resolve_model_field_name,
+)
+from . import utils
 
 _type = type
 
-def type(model, *, fields=None, types=None, **kwargs):
+def get_type_attr(type_, field_name):
+    attr = getattr(type_, field_name, UNSET)
+    if utils.is_unset(attr):
+        attr = getattr(type_, '__dataclass_fields__', {}).get(field_name, UNSET)
+    return attr
+
+def get_field(django_type, field_name, field_annotation=None):
```

Comment:
Explaining the blocks in this function would be helpful

Reply:
I agree. This is one of the most magical function. I added few comments here.